packages/dds/sequence/src/intervalIndex/overlappingSequenceIntervalsIndex.ts

```diff
@@ -16,15 +16,16 @@ import {
 	SequenceInterval,
 	createPositionReferenceFromSegoff,
 } from "../intervals";
+import { SharedString } from "../sharedString";
 import { SequenceIntervalIndexes } from "./sequenceIntervalIndexes";
 import { OverlappingIntervalsIndex } from "./overlappingIntervalsIndex";
 
 class OverlappingSequenceIntervalsIndex
 	extends OverlappingIntervalsIndex<SequenceInterval>
 	implements SequenceIntervalIndexes.Overlapping
 {
-	constructor(client: Client) {
-		super(client, sequenceIntervalHelpers);
+	constructor(sharedString: SharedString) {
+		super(sharedString as unknown as { client: Client } as unknown as Client, sequenceIntervalHelpers);
```

Comment:
this is still not a correct cast:
`const client = (sharedString as unknown as { client: Client }).client;` this will get you the client property off the shared string, which is what you are trying to